### ISSUE TITLE
[js][es6] Invalid generation of static function with name `length`

### ISSUE THREAD

**Issue by RealyUniqueName** [HaxeFoundation/haxe#8185]
```haxe
class Main {
	static function main() trace(length());

	static function length() return 10;
}
```
```
$ haxe -main Main -js test.js -D js-es=6 && node test.js
```
Emits `TypeError: Main.$length is not a function`
Generated js:
```js
// Generated by Haxe 4.0.0-rc.2
(function ($global) { "use strict";
class Main {
	static main() {
		console.log("src/Main.hx:2:",Main.$length());
	}
	static length() {
		return 10;
	}
}
Main.main();
})({});
```

**Comment by nadako** [comment]
Should use `static_field` for static method names there...

**Comment by RealyUniqueName** [comment]
Fixed in 03b63f49ce5c687c8ce5ad6ff9411144dd24f7b0
